x/incentives/keeper/gauge.go

```diff
@@ -281,28 +275,32 @@ func (k Keeper) CreateGroupGauge(ctx sdk.Context, coins sdk.Coins, numEpochPaidO
 	return nextGaugeId, nil
 }
 
-// AddToGaugeRewardsFromGauge transfer coins from groupGaugeId to InternalGaugeId.
```

Comment:
The diff below this line for this section seems to have gotten messy and would probably be easiest to review in split diff view.

The changes made here are:
1. Implement `getWeightBySplittingPolicy` helper
2. Cut out the part of `AddToGaugeRewardsFromGauge` that leans on the now-removed `InternalIds` field

The latter function will be revisited in #6402 once group gauge distribution logic is properly implemented as mentioned in the comment